src/engine/SCons/Taskmaster.py

```diff
@@ -1016,6 +1016,27 @@ def will_not_build(self, nodes, node_func=lambda n: None):
                 to_visit = to_visit | parents
                 pending_children = pending_children - parents
 
+                # Clean up the other branches of side effect
+                for se in node.get_side_effects():
+                    waiting_s_e = se.get_waiting_s_e()
+                    pending_children = pending_children - waiting_s_e
+                    stack = set(waiting_s_e)
+                    if T:
+                        for n in waiting_s_e:
```

Comment:
Minor: could use `stack` instead of `waiting_s_e` here to avoid printing duplicates in the trace.